src/service/gif.js

```diff
@@ -1,9 +1,13 @@
 import api from './config';
 
-export const getGifFeatured = async () =>
+export const getGifFeatured = async (search) =>
   new Promise((resolve, reject) => {
     api
-      .get('/chat/gif/featured')
+      .get(
+        search?.trim() !== ''
+          ? `/chat/gif/search?q=${encodeURIComponent(search.trim())}`
+          : '/chat/gif/featured'
+      )
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new implementation of `getGifFeatured` function now accepts a `search` parameter and constructs the API endpoint based on whether a search query is provided or not. This is a good approach to handle both cases (searching for GIFs and getting featured GIFs) within a single function. However, it's important to ensure that the `search` parameter is properly sanitized before being used in constructing the URL to prevent potential security issues such as XSS (Cross-Site Scripting) attacks. In this case, `encodeURIComponent` is used which is a good practice.

One thing to note here is the error handling. The promise rejection case is not handled. If the API request fails for any reason, the error should be caught and handled appropriately. 

Here's how you can add error handling:

```diff
 import api from './config';
 
 export const getGifFeatured = async (search) =>
    new Promise((resolve, reject) => {
      api
       .get(
         search?.trim() !== ''
           ? `/chat/gif/search?q=${encodeURIComponent(search.trim())}`
           : '/chat/gif/featured'
       )
       .then((res) => {
         resolve(res.data);
       })
+      .catch((error) => {
+        reject(error);
+      });
    });
```
This way, if there's an error with the API request, it will be passed up to the caller of `getGifFeatured`, allowing them to handle the error as needed.


<!-- This is an auto-generated comment by OSS CodeRabbit -->